src/vellum/workflows/state/base.py

```diff
@@ -87,7 +87,9 @@ def __setitem__(self, key: Any, value: Any) -> None:
     def __deepcopy__(self, memo: Any) -> "_SnapshottableDict":
         y: dict = {}
         memo[id(self)] = y
-        for key, value in self.items():
+        # Create a snapshot to avoid "dictionary changed size during iteration"
+        items_snapshot = list(self.items())
```

Comment:
we should fetch a list of keys for each of these instead of snapshotting a list of items